iib/workers/tasks/build_containerized_merge.py

```diff
@@ -0,0 +1,491 @@
+
+    # Prepare request
+    prebuild_info = prepare_request_for_build(
+        request_id,
+        RequestConfigMerge(
+            _binary_image=binary_image,
+            overwrite_target_index_token=overwrite_target_index_token,
+            source_from_index=source_from_index,
+            target_index=target_index,
+            distribution_scope=distribution_scope,
+            binary_image_config=binary_image_config,
+        ),
+    )
+
+    source_from_index_resolved = prebuild_info['source_from_index_resolved']
+    target_index_resolved = prebuild_info['target_index_resolved']
+
+    # Set OPM version
+    Opm.set_opm_version(target_index_resolved)
+    opm_version = Opm.opm_version
+
+    _update_index_image_build_state(request_id, prebuild_info)
+
+    mr_details: Optional[Dict[str, str]] = None
+    local_git_repo_path: Optional[str] = None
+    index_git_repo: Optional[str] = None
+    last_commit_sha: Optional[str] = None
+    output_pull_spec: Optional[str] = None
+    original_index_db_digest: Optional[str] = None
+
+    with tempfile.TemporaryDirectory(prefix=f'iib-{request_id}-') as temp_dir:
+        # Get Git repository information
+        index_git_repo = resolve_git_url(
+            from_index=source_from_index, index_repo_map=index_to_gitlab_push_map or {}
+        )
+        if not index_git_repo:
+            raise IIBError(f"Cannot resolve the git repository for {source_from_index}")
+        log.info(
+            "Git repo for %s: %s",
+            source_from_index,
+            index_git_repo,
+        )
+        token_name, git_token = get_git_token(index_git_repo)
+        branch = prebuild_info['ocp_version']
+
+        # Clone Git repository
+        set_request_state(request_id, 'in_progress', 'Cloning Git repository')
+        local_git_repo_path = os.path.join(temp_dir, 'git', branch)
+        os.makedirs(local_git_repo_path, exist_ok=True)
+
+        clone_git_repo(index_git_repo, branch, token_name, git_token, local_git_repo_path)
+
+        localized_git_catalog_path = os.path.join(local_git_repo_path, 'configs')
+        if not os.path.exists(localized_git_catalog_path):
+            raise IIBError(f"Catalogs directory not found in {local_git_repo_path}")
+
+        # Pull both source and target index.db artifacts and read present bundles
+        set_request_state(
+            request_id,
+            'in_progress',
+            'Retrieving the bundles on index.db from source and target index images',
+        )
+
+        # Get the index.db from ORAS registry for source and target index images
+        source_index_db_ref = get_indexdb_artifact_pullspec(source_from_index_resolved)
+        target_index_db_ref = get_indexdb_artifact_pullspec(target_index_resolved)
+        source_index_db_path = get_oras_artifact(source_index_db_ref, temp_dir)
+        target_index_db_path = get_oras_artifact(target_index_db_ref, temp_dir)
+
```

Comment:
Maybe add a check here to ensure index.db was pulled? Something like https://github.com/release-engineering/iib/blob/main/iib/workers/tasks/build_containerized_rm.py#L170